path/expression.go

```diff
@@ -0,0 +1,163 @@
+package path
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+)
+
+// Expression represents an attribute path with expression steps, which can
+// represent zero, one, or more actual Paths.
+type Expression struct {
```

Comment:
Given the description, I was expecting an `.Append()` method here.

I see there is one on the `ExpressionSteps`, but not on the expression itself.

The reason for me asking, is that I'm trying to imagine how we would take the input for a `schemavalidator`, append it to the `path.Expression` of the attribute, and get to the referred-to attributes.

Reply:
Oh goodness, you are absolutely right that I omitted that method. It's important to be able to merge expressions, especially in plan modifiers and validators. I'll get that pushed up shortly.